src/components/singlEventPageComponents/EventDetails.jsx

```diff
@@ -1,7 +1,150 @@
-import React from "react";
+import { doc, onSnapshot } from "firebase/firestore";
+import Image from "next/image";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import React, { useEffect, useState } from "react";
+
+import { firestore } from "@/lib/firebase/controller";
+
+import Loader from "../loader/Loader";
 
 function EventDetails() {
-    return <div>extract details from eventForm in reusableComponents</div>;
+    const [eventDisplay, setEventDisplay] = useState({});
+
+    const router = useRouter();
+    const { id } = router.query;
```

Comment:
is the url query the only possible way to reach to the event doc ?

Reply:
honestly this one is the first one that came to mind, but i will gladly change into more suitable, Any suggestions ??

Reply:
earlier you fetched each event from the database, to display in cards, each card already has the event id unique to it, you can pass the id to the single page directly without quering it from the url, then pass event title to the url, and if to the event page ~ i believe this could work, so when you click the view event/more button, the dynamic page will take that id as a prop and urk will cleaner ? this is all theoriticall tho